nixos/modules/security/acme.nix

```diff
@@ -185,12 +185,15 @@ in
                   path = [ pkgs.simp_le ];
                   preStart = ''
                     mkdir -p '${cfg.directory}'
-                    chown -R '${data.user}:${data.group}' '${cfg.directory}'
+                    chown root:root '${cfg.directory}'
+                    chmod 755 '${cfg.directory}'
                     if [ ! -d '${cpath}' ]; then
                       mkdir '${cpath}'
                     fi
                     chmod ${rights} '${cpath}'
                     chown -R '${data.user}:${data.group}' '${cpath}'
+                    mkdir -p '${data.webroot}'
+                    chown -R '${data.user}:${data.group}' '${data.webroot}'
```

Comment:
Doesn't this result in problems where `user` and `group` were previously intended for `directory`, but now is being used for `webroot`. A previous setup could be broken due to this change.

In addition, intuitively when configuring ACME it would seem `user` and `group`  are intended for ACME itself. With this change nginx needs to have compatible user permissions with ACME, because the permissions for the webroot directory are being altered for ACME. That doesn't seem right?

Maybe create `${data.webroot}/.acme-challenges/` and set the permissions on that directory. At least existing webroot directories won't be altered that way.